core/message-tags-3.3.md

```diff
@@ -135,9 +135,14 @@ Servers MUST reply with the `ERR_INPUTTOOLONG` (`417`) error numeric if a client
 ## Security considerations
 
 Client-only tags should be treated as untrusted data. They can contain any value
-and are not validated by servers in any way.
+and are not validated by servers in any way. The server MAY normalise values by applying the unescape
+and escape process.
 
-Some specifications may involve servers accepting client-initiated tags without the client-only prefix. Such tag values SHOULD be validated by the server before being sent with any response to a client, unless they are specifically specified to be untrusted data.
+Some specifications may involve servers accepting client-initiated tags without the client-only prefix,
+they SHOULD define a validation process to be performed by the server.
+
+Tags without the client-only prefix or that fail validation MUST be removed by the server before being
+sent with any response to another client.
```

Comment:
This is the same ambiguous wording that was there before.

The specification does not state what "unexpected" is or what "validation" is.


> it feels like it's trying to tell servers to remove all tags without the client-only prefix in all cases

That is exactly what it should do. In the absence of another specification defining a server-only tag that should be relayed from one client to others, the tag is "unexpected" and should be removed. Until such a specification is written, the server should remove all server-only tags.